dora/core/client/fs/src/test/java/alluxio/client/file/MetadataCachingFileSystemTest.java

```diff
@@ -248,49 +249,51 @@ public void createAndRename() throws Exception {
     } catch (FileDoesNotExistException e) {
       // expected exception thrown. test passes
     }
-    assertEquals(2, mFileSystemMasterClient.getStatusRpcCount(NOT_EXIST_FILE));
+    assertEquals(2, mRpcCountingFs.getStatusRpcCount(NOT_EXIST_FILE));
   }
 
   @Test
   public void dropMetadataCacheFile() throws Exception {
     mFs.getStatus(FILE);
-    assertEquals(1, mFileSystemMasterClient.getStatusRpcCount(FILE));
+    assertEquals(1, mRpcCountingFs.getStatusRpcCount(FILE));
     mFs.getStatus(FILE);
-    assertEquals(1, mFileSystemMasterClient.getStatusRpcCount(FILE));
+    assertEquals(1, mRpcCountingFs.getStatusRpcCount(FILE));
     mFs.dropMetadataCache(FILE);
     mFs.getStatus(FILE);
-    assertEquals(2, mFileSystemMasterClient.getStatusRpcCount(FILE));
+    assertEquals(2, mRpcCountingFs.getStatusRpcCount(FILE));
   }
 
   @Test
   public void dropMetadataCacheDir() throws Exception {
     mFs.listStatus(DIR);
-    assertEquals(1, mFileSystemMasterClient.listStatusRpcCount(DIR));
+    assertEquals(1, mRpcCountingFs.listStatusRpcCount(DIR));
     mFs.dropMetadataCache(DIR);
     mFs.listStatus(DIR);
-    assertEquals(2, mFileSystemMasterClient.listStatusRpcCount(DIR));
+    assertEquals(2, mRpcCountingFs.listStatusRpcCount(DIR));
   }
 
   @Test
   public void dropMetadataCacheDirFile() throws Exception {
     mFs.listStatus(DIR);
     mFs.getStatus(FILE);
-    assertEquals(1, mFileSystemMasterClient.listStatusRpcCount(DIR));
-    assertEquals(0, mFileSystemMasterClient.getStatusRpcCount(FILE));
+    assertEquals(1, mRpcCountingFs.listStatusRpcCount(DIR));
+    assertEquals(0, mRpcCountingFs.getStatusRpcCount(FILE));
     mFs.dropMetadataCache(FILE);
     mFs.getStatus(FILE);
-    assertEquals(1, mFileSystemMasterClient.getStatusRpcCount(FILE));
+    assertEquals(1, mRpcCountingFs.getStatusRpcCount(FILE));
     mFs.listStatus(DIR);
-    assertEquals(2, mFileSystemMasterClient.listStatusRpcCount(DIR));
+    assertEquals(2, mRpcCountingFs.listStatusRpcCount(DIR));
   }
 
-  class RpcCountingFileSystemMasterClient extends MockFileSystemMasterClient {
-    RpcCountingFileSystemMasterClient() {
-    }
-
+  class RpcCountingUfsBaseFileSystem extends UfsBaseFileSystem {
```

Comment:
Instead of extending from a real parent class, extend from a `DelegatingFileSystem` which is initialized with a mocked delegate. This ensures calling the non-overridden methods will fail the test, instead of doing unexpected things to the local UFS.

Reply:
Done. And this UfsBaseFS is based on a TemporaryFolder now.